core/services/chainlink/application.go

```diff
@@ -1253,14 +1257,34 @@ func (app *ChainlinkApplication) GetFeedsService() feeds.Service {
 }
 
 // ReplayFromBlock implements the Application interface.
-func (app *ChainlinkApplication) ReplayFromBlock(chainID *big.Int, number uint64, forceBroadcast bool) error {
-	chain, err := app.GetRelayers().LegacyEVMChains().Get(chainID.String())
-	if err != nil {
-		return err
-	}
-	chain.LogBroadcaster().ReplayFromBlock(int64(number), forceBroadcast)
-	if app.Config.Feature().LogPoller() {
-		chain.LogPoller().ReplayAsync(int64(number))
+func (app *ChainlinkApplication) ReplayFromBlock(ctx context.Context, chainFamily string, chainID string, number uint64, forceBroadcast bool) error {
+	switch chainFamily {
+	case relay.NetworkEVM:
+		// TODO: Implement EVM Replay on Relayer instead of using LegacyChains - BCFR-1160
+		chain, err := app.GetRelayers().LegacyEVMChains().Get(chainID)
+		if err != nil {
+			return err
+		}
+		//nolint:gosec // this won't overflow
+		fromBlock := int64(number)
+		chain.LogBroadcaster().ReplayFromBlock(fromBlock, forceBroadcast)
+		if app.Config.Feature().LogPoller() {
+			chain.LogPoller().ReplayAsync(fromBlock)
+		}
+	case relay.NetworkSolana:
+		relayer, err := app.GetRelayers().Get(commontypes.RelayID{
+			Network: relay.NetworkSolana,
+			ChainID: chainID,
+		})
+		if err != nil {
+			return err
+		}
+		err = relayer.Replay(ctx, strconv.FormatUint(number, 10), map[string]any{})
+		if err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("Replay not implemented for chain family: %s", chainFamily)
```

Comment:
I don't want to delay this PR since we are blocking others, but let's follow up by simplifying this to pass through:
```suggestion
		relayer, err := app.GetRelayers().Get(commontypes.RelayID{
			Network: chainFamily,
			ChainID: chainID,
		})
		if err != nil {
			return err
		}
		err = relayer.Replay(ctx, strconv.FormatUint(number, 10), map[string]any{})
		if err != nil {
			return err
		}
```
The implementations will tell us whether they support it or not.

Reply:
Sounds good!